integration/proxy_tunnel_strategy_test.go

```diff
@@ -0,0 +1,422 @@
+		HostID:      uuid.New().String(),
+		NodeName:    Loopback,
+		log:         utils.NewLoggerForTests(),
+	})
+
+	conf := service.MakeDefaultConfig()
+	conf.AuthServers = append(conf.AuthServers, utils.FromAddr(p.lb.Addr()))
+	conf.Token = "token"
+	conf.DataDir = t.TempDir()
+
+	conf.Auth.Enabled = false
+	conf.Proxy.Enabled = false
+	conf.SSH.Enabled = true
+
+	require.NoError(t, node.CreateEx(t, nil, conf))
+
+	t.Cleanup(func() {
+		require.NoError(t, node.StopAll())
+	})
+	require.NoError(t, node.Start())
+
+	if p.node == nil {
+		p.node = node
+	} else {
+		t.Error("node already initialized")
+	}
+}
+
+// makeDatabase boostraps a new teleport db instance.
+// It connects to a proxy via a reverse tunnel going through a load balancer.
+func (p *proxyTunnelStrategy) makeDatabase(t *testing.T) {
+	dbAddr := net.JoinHostPort(Host, strconv.Itoa(ports.PopInt()))
+
+	// setup database service
+	db := NewInstance(InstanceConfig{
+		ClusterName: p.cluster,
+		HostID:      uuid.New().String(),
+		NodeName:    Loopback,
+		log:         utils.NewLoggerForTests(),
+	})
+
+	conf := service.MakeDefaultConfig()
+	conf.AuthServers = append(conf.AuthServers, utils.FromAddr(p.lb.Addr()))
+	conf.Token = "token"
+	conf.DataDir = t.TempDir()
+
+	conf.Auth.Enabled = false
+	conf.Proxy.Enabled = false
+	conf.SSH.Enabled = false
+	conf.Databases.Enabled = true
+	conf.Databases.Databases = []service.Database{
+		{
+			Name:     p.cluster + "-postgres",
+			Protocol: defaults.ProtocolPostgres,
+			URI:      dbAddr,
+		},
+	}
+
+	_, role, err := auth.CreateUserAndRole(p.auth.Process.GetAuthServer(), p.username, nil)
+	require.NoError(t, err)
+
+	role.SetDatabaseUsers(types.Allow, []string{types.Wildcard})
+	role.SetDatabaseNames(types.Allow, []string{types.Wildcard})
+	err = p.auth.Process.GetAuthServer().UpsertRole(context.Background(), role)
+	require.NoError(t, err)
+
+	// start the process and block until specified events are received.
+	require.NoError(t, db.CreateEx(t, nil, conf))
+	t.Cleanup(func() {
+		require.NoError(t, db.StopAll())
+	})
+
+	receivedEvents, err := startAndWait(db.Process, []string{
+		service.DatabasesIdentityEvent,
+		service.DatabasesReady,
+		service.TeleportReadyEvent,
+	})
+	require.NoError(t, err)
+
+	var client *auth.Client
+	for _, event := range receivedEvents {
+		if event.Name == service.DatabasesIdentityEvent {
+			conn, ok := (event.Payload).(*service.Connector)
+			require.True(t, ok)
+			client = conn.Client
+			break
+		}
+	}
+	require.NotNil(t, client)
+
+	// setup a test postgres database
+	postgresDB, err := postgres.NewTestServer(common.TestServerConfig{
+		AuthClient: client,
+		Name:       p.cluster + "-postgres",
+		Address:    dbAddr,
+	})
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		require.NoError(t, postgresDB.Close())
+	})
+	go postgresDB.Serve()
+
+	if p.db == nil {
+		p.db = db
+		p.dbAuthClient = client
+		p.postgresDB = postgresDB
+	} else {
+		t.Error("database already initialized")
+	}
```

Comment:
Same: check for `p.db != nil` at the beginning of the function and change from `t.Error` to `require.Fail`.